### PR TITLE
task(ci): Create docker image for CI

### PR DESCRIPTION
**Pull request by dschom**
## Because

- We are in the process of experimenting with CI pipeline optimizations

## This pull request

- Creates a new docker image that can be used in other CI jobs.
- This base image comes pre-populated with node_modules, dist folders, and l10n repos.

## Issue that this pull request solves

Closes: FXA-660

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Other information (Optional)

There will be follow up PRs to apply this across image to our CI jobs and see if there any performance gains.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -414,6 +432,16 @@ workflows:
                 - /^dockerpush.*/
             tags:
               ignore: /.*/
+
+      - deploy-fxa-ci-image:
+          filters:
+            branches:
+              only:
+                - main
+                - FXA-6260
```
Comment (vbudhram) [inline .circleci/config.yml:441]: This still needed?

**Review by vbudhram — APPROVED** [review]
@dschom I really thought we had a base docker image for fxa that did something similar? Can't seem to find anything in our docs though.

**Comment by chenba** [comment]
> Creates a new docker image that can be used in other CI jobs.

Just curious what CI jobs you meant here.

> I really thought we had a base docker image for fxa that did something similar? Can't seem to find anything in our docs though.

We cache based on yarn.lock.  Is that what you are thinking of?

File: .circleci/config.yml (hunk 1 above)
Comment (dschom) [reply .circleci/config.yml:441]: Nope. That was just for testing. Should have called this out myself. I will remove it.

**Pull request merged**